vm/src/dictdatatype.rs

```diff
@@ -223,6 +240,10 @@ impl<T: Clone> Dict<T> {
                 let mut inner = self.write();
                 // Update existing key
                 if let Some(entry) = inner.entries.get_mut(index) {
+                    let entry = entry.as_mut().unwrap_or_else(|| unsafe {
+                        // SAFETY: `unwrap()` is guaranteed to success because `index` is guaranteed to refer existing entry
+                        std::hint::unreachable_unchecked()
```

Comment:
I'm not sure this is guaranteed - couldn't something happen in between `lookup()` and `write()`?

Reply:
Though I suppose it could still happen before this PR too, but I'd just be hesitant about using `unreachable_unchecked` - there are already cases where dict panics when there's multiple threads accessing one.

Reply:
keeping these future problems as unwraps to fix them later make sense, but `unreachable_unchecked` still panic in Debug. I don't think we can keep everything bare `unwrap` for release mode forever.

Reply:
for this case, then rather than using unwrap_or_else, use unwrap to make panic? I'm not sure about the solution of resolving the multi thread issue. Can i remain it to another issue?

Reply:
Yes, @coolreader18 's comment is correct. Something like `expect("index is changed before retrieving")` will be even better.

Reply:
i chose the message "// The dict was changed since we did lookup. Let's try again." which is used for comment in code